pxtsim/logs.ts

```diff
@@ -190,50 +199,51 @@ namespace pxsim.logs {
                 });
             }
         }
-
+**/
         registerEvents() {
             window.addEventListener('message', (ev: MessageEvent) => {
                 let msg = ev.data;
                 switch (msg.type || '') {
                     case 'serial':
                         const smsg = msg as pxsim.SimulatorSerialMessage;
-                        if (this.dropSim && smsg.sim) {
-                            // drop simulated event since we are receiving real events
-                            return;
-                        } else if (!this.dropSim && !smsg.sim) {
-                            // first non-simulator serial event, drop all previous events
-                            this.clear();
-                            this.dropSim = true;
-                        }
-                        const value = smsg.data || '';
-                        const source = smsg.id || '?';
-                        let theme = source.split('-')[0] || '';
-                        if (!/^[a-z]+$/.test(theme)) theme = 'black';
-                        let buffer = this.serialBuffers[source] || '';
-                        for (let i = 0; i < value.length; ++i) {
-                            switch (value.charCodeAt(i)) {
-                                case 10: //'\n'
-                                    this.appendEntry(source, buffer, theme);
-                                    buffer = '';
-                                    break;
-                                case 13: //'\r'
-                                    break;
-                                default:
-                                    buffer += value[i];
-                                    if (buffer.length > (this.props.maxLineLength || 255)) {
-                                        this.appendEntry(source, buffer, theme);
-                                        buffer = ''
-                                    }
-                                    break;
+                        const sim = !!smsg.sim || false;
+                        if (sim == this.isSim) {
```

Comment:
break instead of nesting

Reply:
Any particular reason why?  This is a bit more concise, and should compile to the same thing.

Reply:
Personal preference to avoid nesting too much. Disregard.

Reply:
ok